### PR TITLE
Enable vertx-websocket extension to handle Quarkus TLS Registry configuration

### PR DESCRIPTION
**Pull request by jamesnetherton**
Relates to #6393.

### CODE REVIEW

**Review by zhfeng — APPROVED** [review]

**Comment by zhfeng** [comment]
Well done!

**Comment by jamesnetherton** [comment]
One of the example projects uncovers an issue that I need to fix (strange it does not happen in the main project test suite).

```
Caused by: jakarta.enterprise.inject.CreationException: Error creating synthetic bean [DBlwsZyrCioOGVZEX1uhn5gHUe0]: jakarta.enterprise.inject.CreationException: Synthetic bean instance for io.quarkus.tls.TlsConfigurationRegistry not initialized yet: io_quarkus_tls_TlsConfigurationRegistry_c3690e7f4866a79befda78c4c24a5b11fe6d5e3a
	- a synthetic bean initialized during RUNTIME_INIT must not be accessed during STATIC_INIT
	- RUNTIME_INIT build steps that require access to synthetic beans initialized during RUNTIME_INIT should consume the SyntheticBeansRuntimeInitBuildItem
	at io.quarkus.tls.TlsConfigurationRegistry_DBlwsZyrCioOGVZEX1uhn5gHUe0_Synthetic_Bean.doCreate(Unknown Source)
	at io.quarkus.tls.TlsConfigurationRegistry_DBlwsZyrCioOGVZEX1uhn5gHUe0_Synthetic_Bean.create(Unknown Source)
	at io.quarkus.tls.TlsConfigurationRegistry_DBlwsZyrCioOGVZEX1uhn5gHUe0_Synthetic_Bean.create(Unknown Source)
	at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:119)
	at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:38)
	at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:35)
	at io.quarkus.arc.impl.LazyValue.get(LazyValue.java:32)
	at io.quarkus.arc.impl.ComputingCache.computeIfAbsent(ComputingCache.java:69)
	at io.quarkus.arc.impl.ComputingCacheContextInstances.computeIfAbsent(ComputingCacheContextInstances.java:19)
	at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:35)
	at io.quarkus.tls.TlsConfigurationRegistry_DBlwsZyrCioOGVZEX1uhn5gHUe0_Synthetic_Bean.get(Unknown Source)
	at io.quarkus.tls.TlsConfigurationRegistry_DBlwsZyrCioOGVZEX1uhn5gHUe0_Synthetic_Bean.get(Unknown Source)
	at io.quarkus.arc.impl.InstanceImpl.getBeanInstance(InstanceImpl.java:325)
	at io.quarkus.arc.impl.InstanceImpl.getInternal(InstanceImpl.java:309)
	at io.quarkus.arc.impl.InstanceImpl.get(InstanceImpl.java:190)
	at io.quarkus.arc.InjectableInstance.orElse(InjectableInstance.java:71)
	at io.quarkus.arc.InjectableInstance.orNull(InjectableInstance.java:82)
	at org.apache.camel.quarkus.component.vertx.websocket.VertxWebsocketRecorder.httpServerTlsRegistryConfigurationExists(VertxWebsocketRecorder.java:79)
```

File: integration-tests/vertx-websocket/src/test/java/org/apache/camel/quarkus/component/vertx/websocket/it/VertxWebsocketTlsNamedConfigTest.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.QuarkusTestProfile;
+import io.quarkus.test.junit.TestProfile;
+import me.escoffier.certs.Format;
+import me.escoffier.certs.junit5.Certificate;
+import org.apache.camel.quarkus.test.support.certificate.TestCertificates;
+
+@TestCertificates(certificates = {
+        @Certificate(name = "vertx-websocket", formats = {
+                Format.PKCS12, Format.PEM }, password = "changeit") })
```
Comment (ppalaga) [inline integration-tests/vertx-websocket/src/test/java/org/apache/camel/quarkus/component/vertx/websocket/it/VertxWebsocketTlsNamedConfigTest.java:30]: Do we use the PKCS12 format somewhere? I see just the PEM one being used below.

Comment (jamesnetherton) [reply integration-tests/vertx-websocket/src/test/java/org/apache/camel/quarkus/component/vertx/websocket/it/VertxWebsocketTlsNamedConfigTest.java:30]: Yes, in the WS client:

https://github.com/apache/camel-quarkus/blob/main/integration-tests/vertx-websocket/src/test/java/org/apache/camel/quarkus/component/vertx/websocket/it/VertxWebsocketClientSslProvider.java

**Pull request merged**